extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -66,327 +66,364 @@ export const RTC_ERROR_ENUM = {
     }
-    const whitelist = {ERROR_TYPE: true, HREF: true};
-    const macros = {
-      ERROR_TYPE: errorType,
-      HREF: win.location.href,
-    };
-    const url = Services.urlReplacementsForDoc(ampDoc).expandUrlSync(
-        errorReportingUrl, macros, whitelist);
-    new win.Image().src = url;
+    return Promise.resolve(/**@type {rtcResponseDef} */(
+      {error, callout, rtcTime: opt_rtcTime || 0}));
   }
-}
 
-/**
- * Converts a URL into its corresponding shortened callout string.
- * We also truncate to a maximum length of 50 characters.
- * For instance, if we are passed
- * "https://example.com/example.php?foo=a&bar=b, then we return
- * example.com/example.php
- * @param {string} url
- * @return {string}
- * @visibleForTesting
- */
-export function getCalloutParam_(url) {
-  const parsedUrl = parseUrlDeprecated(url);
-  return (parsedUrl.hostname + parsedUrl.pathname).substr(0, 50);
-}
-
-/**
- * For a given A4A Element, sends out Real Time Config requests to
- * any urls or vendors specified by the publisher.
- * @param {!./amp-a4a.AmpA4A} a4aElement
- * @param {!Object<string, !../../../src/service/variable-source.AsyncResolverDef>} customMacros The ad-network specified macro
- *   substitutions available to use.
- * @param {?CONSENT_POLICY_STATE} consentState
- * @return {Promise<!Array<!rtcResponseDef>>|undefined}
- * @visibleForTesting
- */
-export function maybeExecuteRealTimeConfig_(
-  a4aElement, customMacros, consentState) {
-  // TODO(keithwrightbos) - allow pub to override such that some callouts are
-  // still allowed.
-  if (consentState == CONSENT_POLICY_STATE.INSUFFICIENT ||
-      consentState == CONSENT_POLICY_STATE.UNKNOWN) {
-    return;
+  /**
+   * @param {string} errorType Uses the RTC_ERROR_ENUM above.
+   * @param {string} errorReportingUrl
+   */
+  sendErrorMessage(errorType, errorReportingUrl) {
+    if (ERROR_REPORTING_ENABLED || getMode(this.win_).localDev
+        || getMode(this.win_).test) {
+      if (!isSecureUrl(errorReportingUrl)) {
+        dev().warn(TAG, `Insecure RTC errorReportingUrl: ${errorReportingUrl}`);
+        return;
+      }
+      const whitelist = {ERROR_TYPE: true, HREF: true};
+      const macros = {
+        ERROR_TYPE: errorType,
+        HREF: this.win_.location.href,
+      };
+      const url = this.urlReplacements_.expandUrlSync(
+          errorReportingUrl, macros, whitelist);
+      new this.win_.Image().src = url;
+    }
   }
-  const rtcConfig = validateRtcConfig_(a4aElement.element);
-  if (!rtcConfig) {
-    return;
+
+  /**
+   * Converts a URL into its corresponding shortened callout string.
+   * We also truncate to a maximum length of 50 characters.
+   * For instance, if we are passed
+   * "https://example.com/example.php?foo=a&bar=b, then we return
+   * example.com/example.php
+   * @param {string} url
+   * @return {string}
+   */
+  getCalloutParam_(url) {
+    const parsedUrl = parseUrlDeprecated(url);
```

Comment:
parseUrlDeprecated is deprecated (hence the name)... move to the new methodology